### PR TITLE
 Add test for configuring cleaning policies

### PR DESCRIPTION
**Pull request by Deixx**
Rework of https://github.com/Open-CAS/open-cas-linux/pull/762

### CODE REVIEW

File: test/functional/api/cas/cache_config.py [hunk 1]
```diff
@@ -143,6 +143,21 @@ def __eq__(self, other):
...
+                else "default"),
+               "flush max buffers: " +
+               (f"{self.flush_max_buffers} " if self.flush_max_buffers is not None
+                else "default"),
+               "staleness time: " +
+               (f"{self.staleness_time} " if self.staleness_time is not None
+                else "default"),
+               "wake up time: " +
+               (f"{self.wake_up_time} " if self.wake_up_time is not None
+                else "default")]
```
Comment (rafalste) [inline test/functional/api/cas/cache_config.py:158]: All those "default" strings should have trailing space to fit the desired convention.
Or even better, you can put each parameter like this to save a few bytes of code ;)
`[f"activity threshold: {self.activity_threshold or 'default'} ", ...]`

File: test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+        params = FlushParametersAlru(
+            wake_up_time=time_wake_up,
+            staleness_time=time_idle,
+            flush_max_buffers=int(data_to_flush.get_value(Unit.Blocks4096)),
+            activity_threshold=time_idle
+        )
+        TestRun.LOGGER.info(str(params))
+        cache.set_params_alru(params)
+        new_params = cache.get_flush_parameters_alru()
```
Comment (rafalste) [inline test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py:104]: Maybe move this line to the next step?

File: test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        dd = (
+            Dd().input("/dev/zero")
+                .output(core.path)
+                .block_size(Size(1, Unit.MiB))
+                .oflag("direct")
+        )
+        dd.run()
+
+    with TestRun.step("Check core statistics."):
+        core_dirty_before = core.get_dirty_blocks()
```
Comment (rafalste) [inline test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py:76]: This step is actually not checking anything, it just fetches some data. Maybe merge it with the next one, which in fact checks this data against another data?

File: test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            TestRun.LOGGER.error(
+                f"Dirty data statistics are the same despite the ALRU policy was used.\n"
+                f"dirty data before pause: {core_dirty_before}\n"
+                f"dirty data after pause: {core_dirty_after}"
+            )
+        elif core_dirty_before != core_dirty_after + data_to_flush:
+            TestRun.LOGGER.error(
+                f"Number of dirty blocks flushed differs from configured in policy.\n"
+                f"configured dirty data to flush: {data_to_flush}\n"
+                f"currently dirty data flushed: "
```
Comment (rafalste) [inline test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py:152]: Just a suggestion to modify those strings to "expected dirty data to flush" and "actual dirty data flushed" to fit the convention.

File: test/functional/api/cas/cache_config.py (hunk 1 above)
Comment (Deixx) [reply test/functional/api/cas/cache_config.py]: Removed trailing spaces as there is join(" | ") after creating the list

File: test/functional/api/cas/cache_config.py [hunk 5]
```diff
@@ -180,12 +189,13 @@ def __eq__(self, other):
...
-            ret += f"acp flush max buffers value: {self.flush_max_buffers} "
-        if self.wake_up_time is not None:
-            ret += f"acp wake up time value: {self.wake_up_time.total_milliseconds()}"
-        return ret
+        ret = ["flush max buffers: " +
+               (f"{self.flush_max_buffers} " if self.flush_max_buffers is not None
+                else "default"),
+               "wake up time: " +
+               (f"{self.wake_up_time} " if self.wake_up_time is not None
+                else "default")]
```
[the remark this answers is not in the document]
Comment (Deixx) [reply test/functional/api/cas/cache_config.py]: same

**Review by rafalste — APPROVED** [review]

File: test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_tools.dd import Dd
+from test_utils.os_utils import Udev
+from test_utils.size import Size, Unit
+from test_utils.time import Time
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_cleaning_policy_config():
```
Comment (jfckm) [inline test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py:28]: I think you should split this tests to three separate tests - that way we'll know what failed exactly at first glance

**Review by jfckm — CHANGES_REQUESTED** [review]

File: test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py (hunk 6 above)
Comment (Deixx) [reply test/functional/tests/lazy_writes/cleaning_policy/test_cleaning_params.py:28]: This is such a basic test and should be so easy to analyze that I think it shouldn't require going through test prepare per each cleaning policy.

**Comment by arutk** [comment]
retest this please

**Pull request merged**